internal/host/host.go

```diff
@@ -728,6 +736,31 @@ func (m *Manager) ReportValidationFailedMetrics(ctx context.Context, h *models.H
 	return nil
 }
 
+func (m *Manager) reportValidationStopSucceedingMetrics(ctx context.Context, vc *validationContext, h *models.Host, newValidationRes map[string][]validationResult) error {
+	log := logutil.FromContext(ctx, m.log)
+	var currentValidationRes map[string][]validationResult
+	if h.ValidationsInfo != "" {
+		if err := json.Unmarshal([]byte(h.ValidationsInfo), &currentValidationRes); err != nil {
+			log.WithError(err).Errorf("Failed to unmarshal validations info from host %s in cluster %s", h.ID, h.ClusterID)
+			return err
+		}
+		for vCategory, vRes := range newValidationRes {
+			for i, v := range vRes {
+				if v.Status == ValidationFailure && currentValidationRes[vCategory][i].Status == ValidationSuccess {
+					m.metricApi.HostValidationChanged(vc.cluster.OpenshiftVersion, h.ClusterID, vc.cluster.EmailDomain, models.HostValidationID(v.ID))
+					eventMsg := fmt.Sprintf("Host validation '%v' that used to succeed is now failing for host %s in cluster %s", v.ID, h.ID, h.ClusterID)
+					m.eventsHandler.AddEvent(ctx, h.ClusterID, h.ID, models.EventSeverityWarning, eventMsg, time.Now())
+				}
+				if v.Status == ValidationSuccess && currentValidationRes[vCategory][i].Status == ValidationFailure {
+					eventMsg := fmt.Sprintf("Host validation '%v' is now fixed for host %s in cluster %s", v.ID, h.ID, h.ClusterID)
```

Comment:
```suggestion
				if v.Status != currentValidationRes[vCategory][i].Status {
					m.metricApi.HostValidationChanged(vc.cluster.OpenshiftVersion, h.ClusterID, vc.cluster.EmailDomain, models.HostValidationID(v.ID))
					eventMsg := fmt.Sprintf("Host validation status '%v' changed, used to %s is now %s  for host %s in cluster %s", v.ID,  v.Status, currentValidationRes[vCategory][i].Status, h.ID, h.ClusterID)
					m.eventsHandler.AddEvent(ctx, h.ClusterID, h.ID, models.EventSeverityWarning, eventMsg, time.Now())
				}

```

Reply:
We are not handling in each change, we are only:
1. reporting metric in case a validation that already passes is now failing.
2. reporting an event when a non working validation got fixed.